src/Datadog.Trace/Tracer.cs

```diff
@@ -109,7 +118,10 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
             // fall back to default implementations of each dependency if not provided
             _agentWriter = agentWriter ?? new AgentWriter(new Api(Settings.AgentUri, delegatingHandler: null, Statsd), Statsd);
 
-            _scopeManager = scopeManager ?? new AsyncLocalScopeManager();
+            // Central Datadog.Trace version independent scope storage
+            CoreStorage.ScopeManager = scopeManager ?? new AsyncLocalScopeManager();
```

Comment:
In the use case where our automatic instrumentation gets the Tracer via `Tracer.Instance` and then the user code gets the Tracer (different version) via `Tracer.Instance`, I think the code as-is will replace the `CoreStorage.ScopeManager` with a new scope manager when the second call is made. Here's my suggestion:

```
_scopeManager = scopeManager ?? CoreStorage.ScopeManager ?? new AsyncLocalScopeManager();
CoreStorage.ScopeManager = _scopeManager;
```

Reply:
Should we only ever set `CoreStorage.ScopeManager` once and have an override method?

Reply:
Do we even need to replace `_scopeManager` with the static `CoreStorage.ScopeManager`? What does that get us?

Reply:
We need to store the ScopeManager ref in the shared `Datadog.Trace.Core` assembly between disparate `Datadog.Trace` versions in order to have manual/automatic traces properly nest.